add low resolution tests

DeniseWorthen opened the pull request:
### Description of changes

- Adds  tests to replicate existing c96-1/4 deg tests (decomp, threads etc) at c96-1deg for the coupled model.

- Changes default to be a 1d run. Replaces 2d/3d/restart test with a 12h/1d/restart test. Restart test is currently implemented at c96mx025 resolution.

- Changes mediator restart directory to RESTART. Mediator restart ``case_name`` is changed from "ufs.s2s" to "ufs.cpld"

- Updates MOM6 and FV3 to match current ufs-weather

### Specific notes

Issues Fixed (include github issue #):

ufs-weather Issue [#222](https://github.com/ufs-community/ufs-weather-model/issues/222)
Issue #83 
Dependent PR #204 

Are changes expected to change answers?
- new baselines required for all tests
- tests which replicate current baselines (c384mx025) were shown to be b4b in PR #204

Specific changes:
- cold nems.configure files are removed
- cpld_control_c192 (c192mx050), cpld_ca (c96mx100 Cellular Automata) are added in addition to c96/mx100 version of existing tests. 
- All tests run with either ``fv3_conf/cpld_control_run.IN`` or ``fv3_conf/cpld_bmark_run.IN`` 


Testing performed:
- machines: hera,orion,dell-p3
- details:

Hashes used for testing:

- NEMS: f102e36
- CMEPS: 4d50adf
- FV3: b955f81
- MOM6: 7a76a13
- CICE: ac617cd
- WW3: 9c22b13
- FMS:f61416f
- stochastic_physics: 6eac84

Co-authored by:

DeniseWorthen committed:
add restart_coszen to namelist; default true

DeniseWorthen committed:
fixed directory for mesh ice file

DeniseWorthen committed:
update NEMS for correct path to forapps/ufs in

cice6 component make

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
update to CMEPS

DeniseWorthen committed:
Merge branch 'develop' into feature/cice6

DeniseWorthen committed:
update NEMS

DeniseWorthen committed:
update CICE5

DeniseWorthen committed:
remove CICE5 and nems RT tests
make all compile statements CICE6

DeniseWorthen committed:
remove cached CICE_SRC

DeniseWorthen committed:
start adding variables to allow configurable resolution in for ice and ocean

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
more resolution variables for tests

DeniseWorthen committed:
more files

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
update CICE for latest Consortium master

DeniseWorthen committed:
update NEMS submodule for mom6 and cice6 interfaces

aerorahul committed:
feature/cice6-cmake (#43)

* move submodules around

Change-Id: Ibdb7216abe1e50df41fce0826f450cbdcfe101ef

* cmake required changes

Change-Id: I54ccc0dc678b195ea9a5d18dfad12e2ed3c179fe

* forgot the NetCDF finder

Change-Id: Ide433b5a16c180b7576e73f08484dfc01430399f

* update the list of files in MOM6 following latest merge

Change-Id: Ib7d5d5f6f3fbc18d026b8400de371a8001de69d8

DeniseWorthen committed:
remove appBuilder files and compsets subdirectory

DeniseWorthen committed:
remove nems.configure files used with NEMS mediator; add CPLMODE
for setting coupling_mode using CMEPS mediator to either nems_orig
or nems_frac

DeniseWorthen committed:
remove create_baseline.py

DeniseWorthen committed:
remove nems tests; add wcoss_dell_p3 compile

DeniseWorthen committed:
more mods for cice6

add frac_grid default to nml, fix compile for cice6, fix file
name in one of the tests

DeniseWorthen committed:
add wcoss_phase3 files

DeniseWorthen committed:
fix use of NIGLOBAL in templates

DeniseWorthen committed:
update CICE6 w/ compile fix for fswthru

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
fix some variables in tests

DeniseWorthen committed:
changes to mom_input templates

add MOM input template for waves at 1/2 deg
add CHECKSUM=false setting to all mom templates

DeniseWorthen committed:
set restart_ext false by default

DeniseWorthen committed:
update CICE-interface for comp_ice.backend fixes

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
cherry pick updates

DeniseWorthen committed:
update CICE

DeniseWorthen committed:
fails at resource allocation

DeniseWorthen committed:
two fixes for 1/2 deg

DeniseWorthen committed:
updates for mx050 tests

DeniseWorthen committed:
update CICE

DeniseWorthen committed:
Merge branch 'feature/rt050' of https://github.com/DeniseWorthen/ufs-s2s-model into feature/rt050

DeniseWorthen committed:
cherry pick updates from rt050 feature branch

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
update FMS,FV3,stoch

DeniseWorthen committed:
update NEMS

DeniseWorthen committed:
Merge branch 'develop' into feature/cice6

DeniseWorthen committed:
update submodules

DeniseWorthen committed:
Merge branch 'develop' into feature/rt050

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge branch 'develop' into feature/rt050

DeniseWorthen committed:
adding c96/1deg ocean test

DeniseWorthen committed:
remove unfixed conflict in ice_in_template

DeniseWorthen committed:
Merge remote-tracking branch 'origin/feature/rt050' into feature/cice6

DeniseWorthen committed:
remove 35 day tests (old form)

DeniseWorthen committed:
update MOM for statediagnose branch

DeniseWorthen committed:
update mom6

DeniseWorthen committed:
update MOM

DeniseWorthen committed:
update cice for new develop branch

DeniseWorthen committed:
Revert "update cice for new develop branch"

This reverts commit 901705dca76335aa36e9301431638d8ea1d51e2b.

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into feature/cice6

DeniseWorthen committed:
update MOM6 for statediagnose w/ setting dbug flag in nems.configure

DeniseWorthen committed:
update parm w/ cap debug flags

DeniseWorthen committed:
update CICE submodule, add new namelist variables

DeniseWorthen committed:
update mesh file name; update cice submodule

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
point back to NOAA-EMC MOM6

DeniseWorthen committed:
use MOM6 dev/emc

DeniseWorthen committed:
fix MOM6 branch

DeniseWorthen committed:
add setting for coldstart restart reading

DeniseWorthen committed:
remove unused tests, fix mesh name

DeniseWorthen committed:
re-set 35d conf and tests to defaults

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge remote-tracking branch 'origin/develop' into feature/cice6

DeniseWorthen committed:
specify special baseline area for cice6 tests

DeniseWorthen committed:
move resolution dependent items to new branch

new branch feature/rtres has been created to contain items
used to run rts for new resolutions; removed from this branch
in order to make cleaner commit to s2s

DeniseWorthen committed:
update CICE submodule

DeniseWorthen committed:
Merge branch 'feature/cice6' of https://github.com/DeniseWorthen/ufs-s2s-model into feature/cice6

DeniseWorthen committed:
update ice_in_template for cice6

DeniseWorthen committed:
relocate restart_coszen

DeniseWorthen committed:
update CICE

DeniseWorthen committed:
update CICE submodule and move dbug flag settings
in nems.configure

DeniseWorthen committed:
update the emc/develop for CICE

DeniseWorthen committed:
update CICE to emc/develop

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
update to latest FV3

DeniseWorthen committed:
update parm

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into feature/cice6

DeniseWorthen committed:
back to compilefix branch w/ debug fix icepack and dell-p3 macro

DeniseWorthen committed:
update submodules

DeniseWorthen committed:
add git templates (#46)

* Issue templates (#165)

* Update issue templates.

* Add a pull request template form (#166)

Co-authored-by: arun chawla <49994787+arunchawla-NOAA@users.noreply.github.com>
Co-authored-by: BinLi-NOAA <bin.li@noaa.gov>

DeniseWorthen committed:
Merge branch 'develop' into feature/cice6

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge branch 'develop' into feature/cice6

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
update CICE to test machine_id and wcoss compiler

DeniseWorthen committed:
update CICE

DeniseWorthen committed:
update to NEMS branch w/ compile change

DeniseWorthen committed:
update CICE, temp baseline areas in rt.sh

DeniseWorthen committed:
update CICE

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into feature/cice6

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into feature/rtres

DeniseWorthen committed:
update to match feature/cice6 branch

DeniseWorthen committed:
changes for low res rts

DeniseWorthen committed:
updates to use ocean lag

DeniseWorthen committed:
update CMEPS

DeniseWorthen committed:
remote dependency in rt.conf for lowres case as test

DeniseWorthen committed:
remove copy-in of coldstart med file if use_coldstart=false

DeniseWorthen committed:
fix messed up line in template

DeniseWorthen committed:
one more template edit

DeniseWorthen committed:
reduce MOM timesteps for mx050

DeniseWorthen committed:
update CMEPS to updcmeps branch

DeniseWorthen committed:
update templates

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
add IC write for MOM6

DeniseWorthen committed:
remove 050 template

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
update CMEPS with Makefile change

DeniseWorthen committed:
update FV3

DeniseWorthen committed:
update NEMS

DeniseWorthen committed:
Merge branch 'develop' into feature/cice6

DeniseWorthen committed:
update NEMS for MOM directory location

DeniseWorthen committed:
update NEMS for right MOM6 location

DeniseWorthen committed:
update CMEPS

DeniseWorthen committed:
update FV3

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
update WW3 w/ cheyenne build

DeniseWorthen committed:
update wave-bmark test PE count and share across ICE and WAV

DeniseWorthen committed:
update orion and p3 default PEs for wav-bmark RT

DeniseWorthen committed:
update CICE

DeniseWorthen committed:
update CICE to emc/develop

DeniseWorthen committed:
update .IN files for cmeps only

DeniseWorthen committed:
remove MEDCOMP instances

DeniseWorthen committed:
point to temporary cice6 baseline on orion

DeniseWorthen committed:
fix typo in wave pets for orion

DeniseWorthen committed:
commit logs from Orion test-baseline creation

DeniseWorthen committed:
update FV3, switch to NOAA-EMC for CICE

DeniseWorthen committed:
update FV3, CICE

DeniseWorthen committed:
Merge remote-tracking branch 'origin/feature/cice6' into feature/rtres

DeniseWorthen committed:
update MOM6

DeniseWorthen committed:
fix existing cold run sequence to give non-zero values for
ocean at start of cold2

DeniseWorthen committed:
Merge remote-tracking branch 'origin/feature/cice6' into feature/rtres

DeniseWorthen committed:
add back 1deg MOM template

DeniseWorthen committed:
update low res tests

DeniseWorthen committed:
ice-wave pes separate

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
update WW3

DeniseWorthen committed:
add back pes for c096 and c192; remove MEDCOMP

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
update to match jun's onestep coldstart

DeniseWorthen committed:
more changes from jun's cmepsic branch

DeniseWorthen committed:
update to jun's 1-step branch

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
update to 1-step coldstart commit

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
remove cold parm files and 1step cold test

DeniseWorthen committed:
explicitly set resolutions

change default_vars to be c96/mx100
current tests are set as c96/mx025 and c384/mx025

DeniseWorthen committed:
fix duplication in 35d tests

DeniseWorthen committed:
more consolidation of default settings

DeniseWorthen committed:
more consolidation

DeniseWorthen committed:
fix temporary rt.conf

DeniseWorthen committed:
fix fhmax for 1d and 3d test

DeniseWorthen committed:
change default tasking names

DeniseWorthen committed:
fix restart_interval for two tests

DeniseWorthen committed:
All RTs pass on hera

DeniseWorthen committed:
remove CNTLMED_DIR; add s2s_control and 2 tests

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
remove temporary file

DeniseWorthen committed:
add setting to use frac input but not run frac grid

DeniseWorthen committed:
back out removal of CNTLMED_DIR and changes to rt_utils

DeniseWorthen committed:
add wave parameters to MOM_input, false by default

DeniseWorthen committed:
change default for mom6 reestart repro add more s2s tests

DeniseWorthen committed:
fix cut/paste error

DeniseWorthen committed:
3 test failures:
 threads need previous thread resources
 bm1d with waves need new baseline created w/ the wave template
 debug test failure needs investigation

DeniseWorthen committed:
fix for 2thread and debug test failures; remove unneeded setting for high freq output

DeniseWorthen committed:
debug and 2thread tests now pass

DeniseWorthen committed:
add c384 test

DeniseWorthen committed:
add 3d and restart test

DeniseWorthen committed:
add c384 test compared against matching current test; update rt.conf

DeniseWorthen committed:
add bmark tests

DeniseWorthen committed:
fixes to get s2s_control_c384 to repro existing test

DeniseWorthen committed:
s2s control now repros existing 2d/384 test

DeniseWorthen committed:
fixing bmark and bmark wave; white space changes

DeniseWorthen committed:
s2s_bmark repros cpld_fv3_ccpp_384_mom6_cice_cmeps_1d_bmark_rt

DeniseWorthen committed:
need to retain original cpl_dflt for use by 384

DeniseWorthen committed:
annotated rt.conf

DeniseWorthen committed:
s2s bmark wave test now repros (but neither is using waves because
of the wave template bug).
Modfify descriptive text in the tests

DeniseWorthen committed:
remove default values fhmax and fdiag where possible

DeniseWorthen committed:
add 35d tests

DeniseWorthen committed:
fix else statement

DeniseWorthen committed:
remove suite file copy-in

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
trying to get 35d tests working

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
remove fv3_slurm (again)

DeniseWorthen committed:
s2s_bmark_wave repros existing ww3_1d_bmark BL

DeniseWorthen committed:
use correct mom template for waves

DeniseWorthen committed:
more 35d test mods

DeniseWorthen committed:
cherry pick updates from defaultres branch to continue testing

DeniseWorthen committed:
update w/ changes from defaultres

DeniseWorthen committed:
make default 1d instead of 2d; change restart test to be a restart from 12h run

DeniseWorthen committed:
change directory mediator restarts write to

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
fix number of MOM restart files in s2s tests
for restart test copy in med restart from RESTART

DeniseWorthen commented:
@MinsukJi-NOAA @junwang-noaa 

I'm having two problems w/ the s2s_restart test and I need  a little help. I've set up  s2s_control (1day), s2_control_12h (12h) and s2s_restart (a 12h run, dependent to the 12h test). The s2s_restart test compares to the s2s_control baseline. 

1. If I try to create a baseline ( ./rt.sh -cefk > output 2>&1), the s2s_restart test tries to run. Since the restart test doesn't have a baseline, I thought that this test is skipped if the baseline is getting created. The run directory for this step is here:
/work/noaa/stmp/dworthen/stmp/dworthen/S2S_RT/rt_242741

2. The restart test in the above directory does seem to be copying in the right dependent run input (from ../s2s_control_12h). But the run fails with an error about time in the fv3_cap.F90: 

20201001 232020.424 WARNING          PET000 ESMCI_Clock.C:1705 ESMCI::Clock::validate() timeStep equals zero.
20201001 232020.433 ERROR            PET000 ESMCI_Clock.C:373 ESMCI::Clock::set() Wrong data value  - Internal subroutine call returned Error
20201001 232020.433 ERROR            PET000 ESMF_Clock.F90:1695 ESMF_ClockSet() Wrong data value  - Internal subroutine call returned Error
20201001 232020.436 ERROR            PET000 fv3_cap.F90:546 Wrong data value  - Passing error in return code
20201001 232020.436 INFO             PET000 Finalizing ESMF

I must have something wrong in the s2s_restart/model.configure or input.nml but I can't spot what it is.

The build directory is here: /work/noaa/marine/dworthen/S2S_lowres/tests

junwang-noaa commented:
For 1), please make sure that the 5th column of s2s_restart test in rt.conf
is empty, it should not have "fv3".

I can't see your code, please open read permission:
Orion-login-3[13] junwang$ ls /work/noaa/marine/dworthen/S2S_lowres/tests
ls: cannot access /work/noaa/marine/dworthen/S2S_lowres/tests: Permission
denied

On Fri, Oct 2, 2020 at 8:40 AM Denise Worthen <notifications@github.com>
wrote:

> @MinsukJi-NOAA <https://github.com/MinsukJi-NOAA> @junwang-noaa
> <https://github.com/junwang-noaa>
>
> I'm having two problems w/ the s2s_restart test and I need a little help.
> I've set up s2s_control (1day), s2_control_12h (12h) and s2s_restart (a 12h
> run, dependent to the 12h test). The s2s_restart test compares to the
> s2s_control baseline.
>
>    1.
>
>    If I try to create a baseline ( ./rt.sh -cefk > output 2>&1), the
>    s2s_restart test tries to run. Since the restart test doesn't have a
>    baseline, I thought that this test is skipped if the baseline is getting
>    created. The run directory for this step is here:
>    /work/noaa/stmp/dworthen/stmp/dworthen/S2S_RT/rt_242741
>    2.
>
>    The restart test in the above directory does seem to be copying in the
>    right dependent run input (from ../s2s_control_12h). But the run fails with
>    an error about time in the fv3_cap.F90:
>
> 20201001 232020.424 WARNING PET000 ESMCI_Clock.C:1705
> ESMCI::Clock::validate() timeStep equals zero.
> 20201001 232020.433 ERROR PET000 ESMCI_Clock.C:373 ESMCI::Clock::set()
> Wrong data value - Internal subroutine call returned Error
> 20201001 232020.433 ERROR PET000 ESMF_Clock.F90:1695 ESMF_ClockSet() Wrong
> data value - Internal subroutine call returned Error
> 20201001 232020.436 ERROR PET000 fv3_cap.F90:546 Wrong data value -
> Passing error in return code
> 20201001 232020.436 INFO PET000 Finalizing ESMF
>
> I must have something wrong in the s2s_restart/model.configure or
> input.nml but I can't spot what it is.
>
> The build directory is here: /work/noaa/marine/dworthen/S2S_lowres/tests
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-s2s-model/pull/194#issuecomment-702711286>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TO6C6B5C5MDPKF3X6TSIXC3NANCNFSM4RXKX5AQ>
> .
>

DeniseWorthen commented:
I've changed the permissions.

I will remove the fv3; Thanks.

MinsukJi-NOAA commented:
```
jminsuk@Orion-login-2:s2s_restart$ diff model_configure ../s2s_control_12h/model_configure 
21,22c21,22
< restart_interval:        24
< fhrot:                   12
---
> restart_interval:        12
> fhrot:                   0
jminsuk@Orion-login-2:s2s_restart$ diff input.nml ../s2s_control_12h/input.nml 
48c48
<        make_nh = .F.
---
>        make_nh = .T.
71c71
<        na_init = 0
---
>        na_init = 1
83c83
<        external_ic = .F.
---
>        external_ic = .T.
86,87c86,87
<        nggps_ic = .F.
<        mountain = .T.
---
>        nggps_ic = .T.
>        mountain = .F.
101c101
<        warm_start = .T.
---
>        warm_start = .F.
313c313
<          input_filename = 'r'
---
>          input_filename = 'n'
```

MinsukJi-NOAA commented:
Since s2s_control_12h worked, I am comparing model_configure and input.nml between s2s_restart and s2s_control_12h

MinsukJi-NOAA commented:
Denise, can you change the permission for log_orion.intel as well?

```
cd log_orion.intel/
-bash: cd: log_orion.intel/: Permission denied
```

DeniseWorthen commented:
Geez I might have messed up. I changed the permissions but then I tested again with the fv3 removed from that column to see that it does fix the first problem (trying to run the restart test during baseline generation). 

I then started it up another time to verify that the restart test can run and compare. So I've changed the permissions again (maybe the newly generated files have the old permissions?). Now when I try to actually run the restart test (./rt.sh -efk >output 2>&1 &) I'm left only with the 2nd problem (some sort of clock issue where the run just fails).

junwang-noaa commented:
Denise, please set FHMAX=24
in /work/noaa/marine/dworthen/S2S_lowres/tests/tests/s2s_restart

On Fri, Oct 2, 2020 at 11:08 AM Denise Worthen <notifications@github.com>
wrote:

> Geez I might have messed up. I changed the permissions but then I tested
> again with the fv3 removed from that column to see that it does fix the
> first problem (trying to run the restart test during baseline generation).
>
> I then started it up another time to verify that the restart test can run
> and compare. So I've changed the permissions again (maybe the newly
> generated files have the old permissions?). Now when I try to actually run
> the restart test (./rt.sh -efk >output 2>&1 &) I'm left only with the 2nd
> problem (some sort of clock issue where the run just fails).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-s2s-model/pull/194#issuecomment-702788738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TPRULBLUZ2VMNZFUSTSIXUFNANCNFSM4RXKX5AQ>
> .
>

DeniseWorthen commented:
Thanks both of you. The restart test is now running correctly during baseline creation and verification. It isn't passing yet though so I need to double check the settings (particularily MOM6, which we know needs a special setting for restart repro).

DeniseWorthen committed:
fix restart test settings, add CA S2S test

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
remove cpld_* files, change FV3_RESTART_INTERVAL to FV3_RESTART_H for clarity

DeniseWorthen committed:
cherry pick updates from defaultres branch

changes to allow this branch to stay in sync w/ switch of default
resolution

DeniseWorthen committed:
make stagger A the default for 1deg MOM

DeniseWorthen committed:
consistency changes w/ default res branch

DeniseWorthen commented:
I am not able to pass the restart test using c96-1deg. FV3 is not reproducing on restart for 5 fields: atmImp_Faxa_snow, atmImp_Sa_pbot, atmImp_Sa_shum, atmImp_Sa_tbot, atmImp_Sa_z. In all cases, there are 9 points on the FV3 tile3 that do not reproduce on restart.

junwang-noaa commented:
You are using the non-fractional grid ICs, right?

On Mon, Oct 5, 2020 at 4:56 PM Denise Worthen <notifications@github.com>
wrote:

> I am not able to pass the restart test using c96-1deg. FV3 is not
> reproducing on restart for 5 fields: atmImp_Faxa_snow, atmImp_Sa_pbot,
> atmImp_Sa_shum, atmImp_Sa_tbot, atmImp_Sa_z. In all cases, there are 9
> points on the FV3 tile3 that do not reproduce on restart.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-s2s-model/pull/194#issuecomment-703884079>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TPUPWODHKUTEBV6S6TSJIXIVANCNFSM4RXKX5AQ>
> .
>

DeniseWorthen commented:
That is right. I am testing c96/mx100 in non-frac mode. The same test in frac-mode shows an additional 2 fields which do not reproduce: atmImp_Faxa_lwdn and atmImp_Faxa_lwnet. 

In both cases I'm reporting the field that is sent to the mediator (atmImp) from FV3 at restart.

DeniseWorthen committed:
remove RESTART from CNTLMED

DeniseWorthen commented:
I now have a test branch for doing the restart test for non-frac c384mx025 using the new 1d/12h and 12 restart setup. 
Using the original input FV3 input (ie, not the newly generated C384mx025 input which is frac_grid capable),  I find that FV3 is reproducing on restart.

DeniseWorthen committed:
white space changes

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
update template and remove dumpfreq calculation from edit_inputs

DeniseWorthen committed:
remove dumpfreq references

DeniseWorthen committed:
one more change to sync w/ defaultres

DeniseWorthen committed:
syncing w/ restart branch

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen committed:
remove THERMO_SPAN, white space

DeniseWorthen committed:
retain tasking for c96mx025 tests, remove "RT-Baseline" from baseline area name

DeniseWorthen committed:
change s2s to cpld

DeniseWorthen committed:
reset tests for 1d default

DeniseWorthen committed:
add cpld_restart at c96mx100 resolution. this test is not current used

DeniseWorthen committed:
add c96mx025 restart test to rt.conf

DeniseWorthen committed:
fix issues found during trial baseline creation

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
manually update submodules

FV3 is updated to ufs-weather PR#217
CICE,MOM6 and WW3 are updated to match ufs-s2s

DeniseWorthen committed:
add rt related changes from stampede

DeniseWorthen committed:
Merge remote-tracking branch 'upstream/develop' into develop

DeniseWorthen committed:
Merge branch 'develop' into feature/rtres

DeniseWorthen commented:
This will also update NEMS to [f102e36](https://github.com/NOAA-EMC/NEMS/tree/develop). This should be OK since that the NEMS commit states it should happen after weather PR [213](https://github.com/ufs-community/ufs-weather-model/pull/213) which has been closed.

DeniseWorthen committed:
change two remaining references to s2s

DeniseWorthen committed:
update baseline directory

DeniseWorthen committed:
add orion logs

DeniseWorthen committed:
add dell-p3 logs

DeniseWorthen committed:
add hera logs

junwang-noaa commented on tests/tests/cpld_ca:
```diff
@@ -0,0 +1,65 @@
+export CA_SGS=.T.
+export CA_GLOBAL=.T.
+
+export FV3_RUN=cpld_control_run.IN
```
It looks to me the input.nml for this test is still the default: export INPUT_NML=input.mom6_ccpp.nml.IN under parm and[ input.mom6_ccpp.nml.IN ](https://github.com/DeniseWorthen/ufs-s2s-model/blob/feature/rtres/parm/input.mom6_ccpp.nml.IN) still has:
do_ca          = .false.
       ca_sgs         = .false.
should it have: 
       do_ca          = @[DO_CA]
       ca_sgs         = @[CA_SGS]
       ca_global      = @[CA_GLOBAL]

DeniseWorthen committed:
fix for ca test

make do_ca,ca_sbs and ca_global as variables in input.mom6_ccpp.nml.IN
white space in rt.conf

DeniseWorthen commented:
I will regenerate the CA baseline test.

I also have a fix for the FV3_RESTART_H issue and have already tested it. I will merge that fix and then re-run all the baselines.

DeniseWorthen committed:
remove use of fv3_restart_h and use restart_n instead

DeniseWorthen committed:
Merge remote-tracking branch 'origin/bugfix/test_restart_n' into feature/rtres

junwang-noaa commented:
Denise, thanks for making the code changes.

DeniseWorthen committed:
add hera logs

DeniseWorthen committed:
add orion logs

DeniseWorthen committed:
add orion logs

DeniseWorthen commented:
All RTs have been re-run. My commit message on the last one mistakenly said 'orion logs'. It is the p3 logs.

junwang-noaa reviewed (approved):
This is a lot work in the two PRs (194 and 204). Thanks for the nice work!

JessicaMeixner-NOAA reviewed (approved):
Looks great! Thanks for adding the cpld CA case and in general just making everything cleaner.
